chpl/chpl-service/src/main/java/gov/healthit/chpl/attestation/manager/AttestationExceptionEmail.java

```diff
@@ -0,0 +1,84 @@
+import gov.healthit.chpl.exception.EmailNotSentException;
+import gov.healthit.chpl.permissions.ResourcePermissions;
+import gov.healthit.chpl.util.AuthUtil;
+import lombok.extern.log4j.Log4j2;
+
+@Component
+@Log4j2
+public class AttestationExceptionEmail {
+        private ChplEmailFactory chplEmailFactory;
+        private ChplHtmlEmailBuilder chplHtmlEmailBuilder;
+        private ResourcePermissions resourcePermissions;
+        private String emailSubject;
+        private String emailBody;
+
+
+        @Autowired
+        public AttestationExceptionEmail(ChplEmailFactory chplEmailFactory, ChplHtmlEmailBuilder chplHtmlEmailBuilder,
+                ResourcePermissions resourcePermissions,
+                @Value("${changeRequest.attestation.exception.subject}") String emailSubject,
+                @Value("${changeRequest.attestation.exception.body}") String emailBody) {
+            this.chplEmailFactory = chplEmailFactory;
+            this.chplHtmlEmailBuilder = chplHtmlEmailBuilder;
+            this.resourcePermissions = resourcePermissions;
+            this.emailSubject = emailSubject;
+            this.emailBody = emailBody;
+        }
+
+        public void send(AttestationPeriodDeveloperException attestationException) throws EmailNotSentException {
+            List<UserDTO> recipients = resourcePermissions.getAllUsersOnDeveloper(
+                    Developer.builder().id(attestationException.getDeveloper().getId()).build());
+
+            chplEmailFactory.emailBuilder()
+                .recipients(recipients.stream()
+                        .map(user -> user.getEmail())
+                        .collect(Collectors.toList()))
+                .subject(emailSubject)
+                .htmlMessage(createExceptionHtmlMessage(attestationException))
+                .sendEmail();
+
+        }
+
+        private String createExceptionHtmlMessage(AttestationPeriodDeveloperException attestationException) {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd, YYYY");
+            return chplHtmlEmailBuilder.initialize()
+                    .heading("Developer Attestations Submission Reopened")
+                    .paragraph("", String.format(emailBody, getActingBody(), formatter.format(attestationException.getExceptionEnd())))
+                    .footer(true)
+                    .build();
+        }
+
+        private String getActingBody() {
+            if (resourcePermissions.isUserRoleAcbAdmin()) {
+                List<CertificationBodyDTO> acbsForUser = resourcePermissions.getAllAcbsForCurrentUser();
+                if (CollectionUtils.isEmpty(acbsForUser)) {
+                    LOGGER.warn("No ACBs were found for the current user " + AuthUtil.getCurrentUser().getSubjectName());
+                    return "an ONC-ACB";
+                }
+                return acbsForUser.get(0).getName();
+            } else if (resourcePermissions.isUserRoleAdmin()) {
+                return "CHPL Admin";
```

Comment:
In the past, when we had the "Surveillance initiated by" field for Surveillance, if the ACB did it the value was "ONC-ACB", and if either ROLE_ADMIN or ROLE_ONC did it the value was just "ONC". I think we should keep that same thing here.

The basic idea is that if someone with ROLE_ADMIN does it, we're acting on behalf of ONC, and so the messaging should indicate that. If you disagree we can bring this up with the PO.